src/library/scala/collection/mutable/Set.scala

```diff
@@ -78,8 +78,14 @@ trait SetOps[A, +CC[X], +C <: SetOps[A, CC, C]]
     *             are removed.
     */
   def filterInPlace(p: A => Boolean): this.type = {
-    for (elem <- this.toList) // scala/bug#7269 toList avoids ConcurrentModificationException
-      if (!p(elem)) this -= elem
+    var list = this.toList // scala/bug#7269 toList avoids ConcurrentModificationException
```

Comment:
If we know the size--and we should--using `toArray` will avoid a huge batch of garbage.

Reply:
Creating an Array requires a classtag though, I've now created an Array[Any] but that ill box for primitives